metricproviders/datadog/datadog.go

```diff
@@ -0,0 +1,166 @@
+		return metricutil.MarkMeasurementError(newMeasurement, err)
+	}
+
+	newMeasurement.Value = newValue
+	newMeasurement.Phase = newStatus
+	finishedTime := metav1.Now()
+	newMeasurement.FinishedAt = &finishedTime
+
+	return newMeasurement
+}
+
+// Resume should not be used with the datadog provider since all the work should occur in the Run method
+func (p *Provider) Resume(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric, measurement v1alpha1.Measurement) v1alpha1.Measurement {
+	p.logCtx.Warn("Datadog provider should not execute the Resume method")
+	return measurement
+}
+
+// Terminate should not be used with the datadog provider since all the work should occur in the Run method
+func (p *Provider) Terminate(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric, measurement v1alpha1.Measurement) v1alpha1.Measurement {
+	p.logCtx.Warn("Datadog provider should not execute the Terminate method")
+	return measurement
+}
+
+// GarbageCollect is a no-op for the datadog provider
+func (p *Provider) GarbageCollect(run *v1alpha1.AnalysisRun, metric v1alpha1.Metric, limit int) error {
+	return nil
+}
+
+func (p *Provider) evaluateResult(result interface{}, metric v1alpha1.Metric) v1alpha1.AnalysisPhase {
+	successCondition := false
+	failCondition := false
+	var err error
+
+	if metric.SuccessCondition != "" {
+		successCondition, err = evaluate.EvalCondition(result, metric.SuccessCondition)
+		if err != nil {
+			p.logCtx.Warning(err.Error())
+			return v1alpha1.AnalysisPhaseError
+		}
+	}
+	if metric.FailureCondition != "" {
+		failCondition, err = evaluate.EvalCondition(result, metric.FailureCondition)
+		if err != nil {
+			return v1alpha1.AnalysisPhaseError
+		}
+	}
+
+	switch {
+	case metric.SuccessCondition == "" && metric.FailureCondition == "":
+		//Always return success unless there is an error
+		return v1alpha1.AnalysisPhaseSuccessful
+	case metric.SuccessCondition != "" && metric.FailureCondition == "":
+		// Without a failure condition, a measurement is considered a failure if the measurement's success condition is not true
+		failCondition = !successCondition
+	case metric.SuccessCondition == "" && metric.FailureCondition != "":
+		// Without a success condition, a measurement is considered a successful if the measurement's failure condition is not true
+		successCondition = !failCondition
+	}
+
+	if failCondition {
+		return v1alpha1.AnalysisPhaseFailed
+	}
+
+	if !failCondition && !successCondition {
+		return v1alpha1.AnalysisPhaseInconclusive
+	}
+
+	// If we reach this code path, failCondition is false and successCondition is true
+	return v1alpha1.AnalysisPhaseSuccessful
+}
+
+func (p *Provider) processResponse(metric v1alpha1.Metric, response dd.Series) (string, v1alpha1.AnalysisPhase, error) {
+	length := response.GetLength()
+	switch {
+	case length < 1:
+		return "", v1alpha1.AnalysisPhaseInconclusive, nil
+	case length == 1:
+		return fmt.Sprintf("%f", *response.Points[0][1]), v1alpha1.AnalysisPhaseSuccessful, nil
```

Comment:
Why is this a successful return with any evaluation?

Reply:
yeah, just fixed that as part of: https://github.com/cabrinha/argo-rollouts/pull/1/commits/a52722d8ffc4d392925b6db6dfbdeaa19fea3aa9